xl2times/utils.py

```diff
@@ -315,6 +315,57 @@ def round_sig(x, sig_figs):
 default_log_name = "log" if default_log_name == "" else default_log_name
 
 
+def _case_insensitive_match(path: str, pattern: str) -> bool:
+    """Do case-insensitive path match. Convert to lowercase first, because
+    case_sensitive parameter in match is not available before Python 3.12.
+    """
+    return PurePath(path.lower()).match(pattern.lower())
+
+
+def is_veda_based(files: list[str]) -> bool:
+    """Determine whether the model follows Veda file structure.
+    This function does not verify file extensions.
+    """
+    marker = "SysSettings.*"
+
+    matches = [file for file in files if _case_insensitive_match(file, marker)]
+
+    if len(matches) == 1:
+        return True
+    elif len(matches) > 1:
+        raise ValueError(f"Only one {marker} expected. Multiple detected: {matches}")
+    else:
+        return False
+
+
+def filter_veda_filename_patterns(files: list[str]) -> list[str]:
+    """Filter files by patterns recognised by Veda.
+    This function does not verify file extensions.
+    """
+    legal_paths = (
+        "BY_Trans.*",
+        "LMA*.*",
+        "Set*.*",
+        "SysSettings.*",
+        "VT_*.*",
+        "SubRES_TMPL/SubRES_*.*",
+        "SuppXLS/Demands/Dem_Alloc+Series.*",
+        "SuppXLS/Demands/ScenDem_*.*",
+        "SuppXLS/ParScenFiles/Scen_Par-*.*",
+        "SuppXLS/Scen_*.*",
+        "SuppXLS/Trades/ScenTrade_*.*",
+    )
+    # Generate a set of fiels that match the patterns
+    filtered = {
+        file
+        for file in files
+        for legal_path in legal_paths
```

Comment:
If this is/gets slow we should consider building a single regex out of `legal_paths`, e.g. `(Set*.*|VT_*.*|...)` and doing a single match for each `file in files`.

Reply:
So far it seems not to have any impact on speed.